index.js

```diff
@@ -258,25 +258,56 @@ HyperDB.prototype.authorized = function (key, cb) {
   if (!cb) cb = noop
   var self = this
 
-  this.heads(function (err, heads) {
+  if (this.key.equals(key)) return process.nextTick(cb, null, true)
+
+  this._tips(key, function (err, heads) {
     if (err) return cb(err)
+    var max = 0
     for (var i = 0; i < heads.length; i++) {
       var head = heads[i]
+      max = Math.max(head.clock.length)
```

Comment:
`max = Math.max(head.clock.length, max)`?

Reply:
classic @ralphtheninja , always lurking in the shadows

Reply:
:grinning: